class-two-factor-core.php

```diff
@@ -99,6 +99,8 @@ public static function add_hooks( $compat ) {
 		add_filter( 'wp_login_errors', array( __CLASS__, 'maybe_show_reset_password_notice' ) );
 		add_action( 'after_password_reset', array( __CLASS__, 'clear_password_reset_notice' ) );
 		add_action( 'login_form_validate_2fa', array( __CLASS__, 'login_form_validate_2fa' ) );
+		add_action( 'login_form_revalidate_2fa', array( __CLASS__, 'login_form_revalidate_2fa' ) );
```

Comment:
Perhaps the feature flag for that could simply be the revalidate time, if it's falsey disable the feature.

Part of my hesitation in merging this as-is is the danger that no followup UI work happens to make this scenario any better.

I think it would be well agreed that the current Two-Factor UI is not optimal and not at all the most user friendly, but as we know, creating a new UI and supporting existing provider is not going to be the easiest task.

Reply:
Yeah, I agree 👍🏻 I don't feel strongly about a flag.